Cannot Install, Run

I've followed your instructions on the README but have not been able to get the demo to run.

I believe that you may have some modules installed globally that you have not mentioned. 
- I installed `typings` globally, but your `package.json` scripts `build` and `build_demo` requires `tsc` to be installed.
- `electron ./demo` fails via the `demo` script as well as via command line.
- Running `node_modules/electron-prebuilt/dist/Electron.app/Contents/MacOS/Electron ./demo` fails with `Error: Cannot find module '/Users/leifwells/electron/angular-electron-master/demo'` (which makes no sense because that is the correct path)

When possible, help me assist you. I'd like to see this project be successful.

Thanks/

I was able to get npm install to work by changing the package.json to:

```
"dependencies": {
    "angular2": "^2.0.0-beta.11",
    "es6-promise": "^3.1.2",
    "es6-shim": "^0.35.0",
    "parse5": "^1.5.1",
    "reflect-metadata": "0.1.2",
    "rxjs": "5.0.0-beta.2",
    "zone.js": "^0.6.4"
  }
```

However, npm start fails with this error message:
src/platform/electron_message_bus.ts(50,16): error TS2339: Property 'onTurnDone' does not exist on type 'NgZone'.
src/platform/electron_renderer_common.ts(39,39): error TS2307: Cannot find module 'angular2/src/router/browser_platform_location'.

The second error looks like a path issue as that file exists at 'angular2/src/router**_/location_**/browser_platform_location'.

--- a/demo/main.ts
+++ b/demo/main.ts
@@ -1,5 +1,5 @@
 import 'reflect-metadata';
-import 'zone.js/dist/zone-microtask'
+import 'zone.js/dist/zone';
 import {bootstrap} from '../dist/main';
 
 import {App} from './component';

--- a/package.json
+++ b/package.json
@@ -7,7 +7,7 @@
     "clean": "rimraf dist",
     "typings-install": "typings install",
     "setup": "npm install --no-optional && npm run typings-install",
-    "setup-demo": "npm install --no-optional angular2@2.0.0-beta.6 es6-promise@^3.0.2 es6-shim@^0.33.3 reflect-metadata@0.1.2 rxjs@5.0.0-beta.0 zone.js@0.5.14",
+    "setup-demo": "npm install --no-optional angular2@2.0.0-beta.12 es6-promise@^3.1.2 es6-shim@0.35.0 reflect-metadata@0.1.2 rxjs@5.0.0-beta.2 zone.js@^0.6.6",
     "build": "npm run clean && tsc",
     "build_demo": "tsc typings/browser.d.ts demo/main.ts demo/app_ui.ts --module commonjs -t es5 --experimentalDecorators --emitDecoratorMetadata",
     "test": "echo Not supported yet",
@@ -29,12 +29,12 @@
   },
   "homepage": "https://github.com/angular/angular-electron#readme",
   "devDependencies": {
-    "electron-prebuilt": "^0.36.7",
-    "typescript": "^1.7.5",
+    "electron-prebuilt": "^0.37.2",
+    "typescript": "^1.8.9",
     "rimraf": "^2.5.2"
   },
   "peerDependencies": {
-    "angular2": "2.0.0-beta.6"
+    "angular2": "2.0.0-beta.12"
   },
   "dependencies": {
     "parse5": "^1.5.1"

--- a/src/platform/electron_app.ts
+++ b/src/platform/electron_app.ts
@@ -1,13 +1,11 @@
 import * as electron from 'electron';
 import {ElectronMessageBus, ElectronMessageBusSink, ElectronMessageBusSource, ELECTRON_READY} from './electron_message_bus';
 import {ELECTRON_APP_APPLICATION_COMMON, ELECTRON_APP_PLATFORM} from './electron_app_common';
-import {NgZone} from 'angular2/src/core/zone/ng_zone';
 import {Type, CONST_EXPR, isPresent} from 'angular2/src/facade/lang';
-import {Provider} from 'angular2/src/core/di';
 import {Parse5DomAdapter} from 'angular2/src/platform/server/parse5_adapter';
-import {APP_INITIALIZER, platform, ComponentRef} from 'angular2/core';
+import {APP_INITIALIZER, platform, ComponentRef, NgZone, Provider} from 'angular2/core';
 import {MessageBus} from 'angular2/src/web_workers/shared/message_bus';
-import {COMPILER_PROVIDERS} from 'angular2/src/compiler/compiler';
+import {COMPILER_PROVIDERS} from 'angular2/compiler';
 
 export const ELECTRON_APP_APPLICATION: Array<any /*Type | Provider | any[]*/> = [
   ELECTRON_APP_APPLICATION_COMMON,

--- a/src/platform/electron_message_bus.ts
+++ b/src/platform/electron_message_bus.ts
@@ -47,7 +47,7 @@ export class ElectronMessageBusSink implements MessageBusSink {
 
   attachToZone(zone: NgZone): void {
     this._zone = zone;
-    this._zone.onTurnDone.subscribe(() => {this._handleOnEventDone()});
+    this._zone.onMicrotaskEmpty.subscribe(() => {this._handleOnEventDone()});
   }
 
   initChannel(channel: string, runInZone: boolean = true): void {

--- a/src/platform/electron_renderer_common.ts
+++ b/src/platform/electron_renderer_common.ts
@@ -1,6 +1,5 @@
 import {CONST_EXPR, IS_DART} from 'angular2/src/facade/lang';
 import {MessageBus} from 'angular2/src/web_workers/shared/message_bus';
-import {NgZone} from 'angular2/src/core/zone/ng_zone';
 import {
   PLATFORM_DIRECTIVES,
   PLATFORM_PIPES,
@@ -13,7 +12,8 @@ import {
   PLATFORM_COMMON_PROVIDERS,
   RootRenderer,
   PLATFORM_INITIALIZER,
-  APP_INITIALIZER
+  APP_INITIALIZER,
+  NgZone
 } from 'angular2/core';
 import {EVENT_MANAGER_PLUGINS, EventManager} from 'angular2/platform/common_dom';
 import {provide, Provider, Injector, OpaqueToken} from 'angular2/src/core/di';
@@ -36,7 +36,7 @@ import {BrowserDomAdapter} from 'angular2/src/platform/browser/browser_adapter';
 import {wtfInit} from 'angular2/src/core/profile/wtf_init';
 import {MessageBasedRenderer} from 'angular2/src/web_workers/ui/renderer';
 import {MessageBasedXHRImpl} from 'angular2/src/web_workers/ui/xhr_impl';
-import {BrowserPlatformLocation} from 'angular2/src/router/browser_platform_location';
+import {BrowserPlatformLocation} from 'angular2/src/router/location/browser_platform_location';
 import {
   ServiceMessageBrokerFactory,
   ServiceMessageBrokerFactory_

--- a/typings.json
+++ b/typings.json
@@ -1,12 +1,9 @@
 {
   "name": "@angular2/electron",
-  "dependencies": {
-    "es6-promise": "github:typed-typings/npm-es6-promise#fb04188767acfec1defd054fc8024fafa5cd4de7"
-  },
   "devDependencies": {},
   "ambientDependencies": {
     "es6-shim": "github:DefinitelyTyped/DefinitelyTyped/es6-shim/es6-shim.d.ts#6697d6f7dadbf5773cb40ecda35a76027e0783b2",
-    "github-electron": "github:DefinitelyTyped/DefinitelyTyped/github-electron/github-electron.d.ts#ecf4bb33465b2a3f6f3a719ebd028575261137a8",
-    "node": "github:DefinitelyTyped/DefinitelyTyped/node/node.d.ts#aee0039a2d6686ec78352125010ebb38a7a7d743"
+    "github-electron": "github:DefinitelyTyped/DefinitelyTyped/github-electron/github-electron.d.ts#f16c4922a7dd6d359a197e4cc5103463ba68cf9e",
+    "node": "github:DefinitelyTyped/DefinitelyTyped/node/node.d.ts#138ad74b9e8e6c08af7633964962835add4c91e2"
   }
 }

